### ISSUE TITLE
Use a custom multi-select for picking currency or filter options

### ISSUE THREAD

**Issue by rjt-rockx** [BanchanArt/banchan#653]
The current one has a really small click/tap target, and is displayed inline rather than as an overlay. This isn't ideal and can take up valuable screen space, and should be refactored into a custom component that uses a listbox instead.

**Comment by zkat** [comment]
I'm working on writing a full-on ComboBox component we can use to replace all our Select and MultiSelect components.

**Comment by zkat** [comment]
This is done by #762
